engine/macros.hpp

```diff
@@ -8,18 +8,15 @@
   std::hex << std::setfill('0') << std::setw(sizeof(decltype(x)) * 2) << x \
            << std::dec
 
-#if KVDK_DEBUG_LEVEL > 0
+#ifndef KVDK_DEBUG_LEVEL
+#pragma GCC warning "KVDK_DEBUG_LEVEL not defined, defaulted to 0"
+#define KVDK_DEBUG_LEVEL 0
+#endif
+
 #define kvdk_assert(cond, msg)                                           \
   {                                                                      \
-    if (!(cond)) {                                                       \
+    if (KVDK_DEBUG_LEVEL > 0 && !(cond)) {                               \
```

Comment:
if `KVDK_DEBUG_LEVEL` is 0, we can write `#define kvdk_assert(cond, msg)`. like `SyncPoint`: 
![image](https://user-images.githubusercontent.com/7602435/163332736-46671b86-ab54-44d9-bdf8-2d9d37a33ea2.png)